### DISCUSSION TITLE
Determine next song

### DISCUSSION THREAD

**Discussion by SoongNoonien** [MusicPlayerDaemon/MPD#1305] [Q&A]
Hi!
Is there a way to determine the next song to be played even when the response of `currentsong` is empty? I'm aware of the `nextsong` and `nextsongid` keywords in the response of `status` but they are not populated when `currentsong` is empty.
I also wondered why `currentsong` is empty when I clear and refill the queue and after a restart of mpd it returns exactly that what I'm am looking for, namely the song which gets played when I issue `play`. Is this behavior intended?

**Comment by MaxKellermann** [comment]
There is no current song if MPD was never asked to select one. Intended maybe - that word is too strong - let's say it's a natural implication of the design.

**Reply by SoongNoonien** [reply]
> There is no current song if MPD was never asked to select one.

Is it possible to ask MPD to select one without starting playback?

> Intended maybe - that word is too strong - let's say it's a natural implication of the design.

Why is it a natural implication of the design that MPD selects a current song on startup and not when the queue is refilled or finished?
